### PR TITLE
Save window geometry and display index

### PR DESCRIPTION
**Pull request by rfomin**

### CODE REVIEW

File: Source/i_video.c [hunk 1]
```diff
@@ -462,6 +465,11 @@ static void I_HandleKeyboardEvent(SDL_Event *sdlevent)
 
 static void HandleWindowEvent(SDL_WindowEvent *event)
 {
+    int i;
+    int x, y;
+    char buf[128];
```
Comment (fabiangreffrath) [inline Source/i_video.c:470]: `[16]` should be sufficient even for a 10000000x10000000px display. :wink:

**Review by fabiangreffrath — CHANGES_REQUESTED** [review]

File: Source/i_video.c [hunk 2]
```diff
@@ -1142,10 +1219,30 @@ static void I_InitGraphicsMode(void)
...
+   blit_rect.w = v_w;
+   blit_rect.h = v_h;
+
+   actualheight = useaspect ? (6 * v_h / 5) : v_h;
+
    SDL_SetWindowMinimumSize(screen, v_w, actualheight);
 
    // [FG] window size when returning from fullscreen mode
-   if (!window_width || !window_height)
+   if (scalefactor * v_w > window_width)
```
[the remark this answers is not in the document]
Comment (rfomin) [reply Source/i_video.c:1245]: `window_width` and `window_height` are config parameters now. They initialized on config load.

File: Source/i_video.c [hunk 3]
```diff
@@ -496,6 +504,28 @@ static void HandleWindowEvent(SDL_WindowEvent *event)
...
+        // update the video_display config variable.
+
+        case SDL_WINDOWEVENT_MOVED:
+            i = SDL_GetWindowDisplayIndex(screen);
+            if (i >= 0)
+            {
+                video_display = i;
+            }
+            
+            SDL_GetWindowPosition(screen, &x, &y);
```
[the remark this answers is not in the document]
Comment (rfomin) [reply Source/i_video.c:519]: I tried SDL_WINDOWEVENT_CLOSE but for some reason, it didn't work. `window_positon` is a config parameter, memory for it is allocated when the config is loaded. I don't think reallocation when the window is moved is an issue, SDL_WINDOWEVENT_MOVED only fires after the move is complete. But this code is ugly, I agree.

[the remark this answers is not in the document]
Comment (fabiangreffrath) [reply Source/i_video.c:518]: Could it get moved to `I_ShutdownGraphics()`?

[the remark this answers is not in the document]
Comment (rfomin) [reply Source/i_video.c]: Yes, it could. Did it.

File: Source/i_video.c [hunk 4]
```diff
@@ -888,7 +905,20 @@ void I_ShutdownGraphics(void)
 {
    if(in_graphics_mode)  // killough 10/98
    {
-      UpdateGrab();      
+      int x, y;
+      char buf[16];
+      int buflen;
+      
+      SDL_GetWindowPosition(screen, &x, &y);
```
Comment (fabiangreffrath) [inline Source/i_video.c:912]: Could you please add a small comment here?

File: Source/i_video.c [hunk 5]
```diff
@@ -979,6 +1009,68 @@ int cfg_aspectratio; // haleyjd 05/11/09: aspect ratio correction
...
+                        "for display #%d!\n", video_display);
+        return;
+    }
+
+    *x = bounds.x + SDL_max((bounds.w - w) / 2, 0);
+    *y = bounds.y + SDL_max((bounds.h - h) / 2, 0);
+}
+
+void I_GetWindowPosition(int *x, int *y, int w, int h)
+{
```
Comment (fabiangreffrath) [inline Source/i_video.c:1031]: Could we make this static?

**Review by fabiangreffrath — APPROVED** [review]
LGTM. Two questions: Chocolate Doom and friends do not save window position, right? I never realized that. And what do the new config keys read if you delete the config file, start the game and close again.

**Comment by rfomin** [comment]
> Chocolate Doom and friends do not save window position, right?

Right. I didn't know either.

> And what do the new config keys read if you delete the config file, start the game and close again.

Default values from `defaults[]` in `m_misc.c`

**Comment by fabiangreffrath** [comment]
> Default values from `defaults[]` in `m_misc.c`

Sure, but `window_position "center"` is replaced by an actual coordinate, right?

**Comment by fabiangreffrath** [comment]
At the risk of repeating myself: Thank you very much!

**Comment by rfomin** [comment]
> `window_position "center"` is replaced by an actual coordinate

Right. `center` is for manual config editing I guess :)

**Pull request merged**